client/credentials_test.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"testing"
+	"strconv"	
+)
+
+func TestCredentials(t *testing.T){
+
+	cred := &Credentials{
+		ClientID          : "tester"
+		AccessToken       : "no-secret"
+	}
+
+	var credentialsTests = []struct {
+		meth string
+		url  string
+		host string
+		port int
+		hdr  string
+		now  int64
+		perr error
+		verr error
+		key  string
+		hash func() hash.Hash
+		rply bool
+	}
+	{
+		{
+			hdr:  `Hawk id="1", ts="1353788437", nonce="k3j4h2", mac="zy79QQ5/EYFmQqutVnYb73gAc/U=", ext="hello"`,
+			hash: sha1.New,
+		},
+	}
+
+	for k, test := range credentialsTests {
+			if test.meth == "" {
+				test.meth = "GET"
+			}
+			if test.url == "" {
+				test.url = "/resource/4?filter=a"
+			}
+			if test.host == "" {
+				test.host = "example.com"
+			}
+			if test.port == 0 {
+				test.port = 8080
+			}
+			if test.now == 0 {
+				test.now = 1353788437
+			}
+			if test.hash == nil {
+				test.hash = sha256.New
+			}
+			if test.key == "" {
+				test.key = "werxhqb98rpaxn39848xrunpaw3489ruxnpa98w4rxn"
+			}
```

Comment:
This:
```go
var credentialsTests = []struct {
      meth string
      url  string
      host string
      port int
      hdr  string
      now  int64
      perr error
      verr error
      key  string
      hash func() hash.Hash
      rply bool
  }
{
      {
          meth: "GET",
          url: "/resource/4?filter=a",
          host: "example.com",
          port: 8080,
          hdr:  `Hawk id="1", ts="1353788437", nonce="k3j4h2", mac="zy79QQ5/EYFmQqutVnYb73gAc/U=", ext="hello"`,
          now: 1353788437,
          hash: sha1.New,
          key: "werxhqb98rpaxn39848xrunpaw3489ruxnpa98w4rxn"
      },
  }

  for k, test := range credentialsTests {
      // ...
  }
```

* Is more compact
* Decreases the [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) of the code
* It is a [declarative syntax](https://en.wikipedia.org/wiki/Declarative_programming), which is easier to read.

@jonasfj thoughts here?